kani-driver/src/get_det_vals.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::fs;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+
+use crate::session::KaniSession;
+use crate::util::alter_extension;
+use anyhow::Result;
+use kani_metadata::HarnessMetadata;
+use serde_json::Value;
+
+impl KaniSession {
+    /// Extract deterministic values from a failing harness.
+    pub fn get_det_vals(&self, file: &Path, harness_metadata: &HarnessMetadata) -> Result<()> {
+        let results_filename = alter_extension(file, "results.json");
+        let det_vals_filename = alter_extension(file, "det_vals.txt");
+
+        {
+            let mut temps = self.temporaries.borrow_mut();
```

Comment:
These shouldn't be temporaries since we want to enable users to run them from their IDEs.